pgmpy/estimators/ConstraintBasedEstimator.py

```diff
@@ -0,0 +1,648 @@
+        structure learning algorithm. Independencies are identified from the
+        data set using a chi-squared statistic with the acceptance threshold of
+        `p_value`. PC identifies a partially directed acyclic graph (PDAG), given
+        that the tested independencies admit a faithful Bayesian network representation.
+        This method returns a BayesianModel that is a completion of this PDAG.
+
+        Parameters
+        ----------
+        p_value: float, default: 0.01
+            A significance level to use for conditional independence tests in the data set.
+            The p_value is the threshold probability of falsely rejecting the hypothesis
+            that variables are conditionally dependent.
+
+            The lower `p_value`, the more likely we are to reject dependencies,
+            resulting in a sparser graph.
+
+        Returns
+        -------
+        model: BayesianModel()-instance
+            An estimate for the BayesianModel for the data set (not yet parametrized).
+
+        Reference
+        ---------
+        Neapolitan, Learning Bayesian Networks, Section 10.1.2, Algorithm 10.2 (page 550)
+        http://www.cs.technion.ac.il/~dang/books/Learning%20Bayesian%20Networks(Neapolitan,%20Richard).pdf
+
+        Examples
+        --------
+        >>> import pandas as pd
+        >>> import numpy as np
+        >>> from pgmpy.estimators import ConstraintBasedEstimator
+        >>> data = pd.DataFrame(np.random.randint(0, 5, size=(2500, 3)), columns=list('XYZ'))
+        >>> data['sum'] = data.sum(axis=1)
+        >>> print(data)
+              X  Y  Z  sum
+        0     3  0  1    4
+        1     1  4  3    8
+        2     0  0  3    3
+        3     0  2  3    5
+        4     2  1  1    4
+        ...  .. .. ..  ...
+        2495  2  3  0    5
+        2496  1  1  2    4
+        2497  0  4  2    6
+        2498  0  0  0    0
+        2499  2  4  0    6
+
+        [2500 rows x 4 columns]
+        >>> c = ConstraintBasedEstimator(data)
+        >>> model = c.estimate()
+        >>> print(model.edges())
+        [('Z', 'sum'), ('X', 'sum'), ('Y', 'sum')]
+        """
+
+        skel, seperating_sets = self.estimate_skeleton(p_value)
+        pdag = self.skeleton_to_pdag(skel, seperating_sets)
+        model = self.pdag_to_dag(pdag)
+        return model
+
+    def estimate_skeleton(self, p_value=0.01):
+        """Estimates a graph skeleton (UndirectedGraph) for the data set.
+        Uses the build_skeleton method (PC algorithm); independencies are
+        determined using a chisquare statistic with the acceptance threshold
+        of `p_value`. Returns a tuple `(skeleton, seperating_sets).
+
+        Parameters
+        ----------
+        p_value: float, default: 0.01
+            A significance level to use for conditional independence tests in
+            the data set. The p_value is the threshold probability of falsely
```

Comment:
@chrisittner This line is not very clear to me. I think mentioning the null hypothesis and the alternate hypothesis here would make things more clear.